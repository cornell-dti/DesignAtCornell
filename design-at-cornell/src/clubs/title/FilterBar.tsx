import React from 'react';
import {
  FilterBarContainer,
  SearchBar,
  SavedButton,
} from '../../components/FilterBarStyles';
import { Filters, SetFilters } from '../../constants/filter-criteria';
import bookmarked from '../../static/images/bookmarked.svg';
import FilterDropdown from './FilterDropdown';

const FilterBar = (props: Props) => {
  return (
    <FilterBarContainer>
      {props.filterList.map((f) => (
        <FilterDropdown key={f.category} {...{ ...f, setPage: props.setPage }} />
      ))}
      <SearchBar
        placeholder="Name, Keywords, Topics, Etc"
        onChange={(event) => {
          props.setSearch(event.target.value);
          props.setPage(1);
        }}
      />
      <SavedButton>
        <img src={bookmarked} alt={'saved clubs'} />
        <p>Saved Clubs</p>
      </SavedButton>
    </FilterBarContainer>
  );
};

export type Props = {
  filterList: { category: string; tags: Filters; setTags: SetFilters }[];
  setPage: (page: React.SetStateAction<number>) => void;
  setSearch: (search: React.SetStateAction<string>) => void;
};

export default FilterBar;
