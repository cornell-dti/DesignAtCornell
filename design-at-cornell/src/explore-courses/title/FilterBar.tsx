import React from 'react';
import {
  FilterBarContainer,
  FilterButton,
} from '../ExploreCoursesStyles';
import downArrow from '../../static/images/down-arrow.png';

const FilterBar = () => (
  <FilterBarContainer>
    <FilterButton>
      <p>Design Areas</p>
      <img src={downArrow} alt={"expand"} />
    </FilterButton>
    <FilterButton>
      <p>Major/Minors</p>
      <img src={downArrow} alt={"expand"} />
    </FilterButton>
    <FilterButton>
      <p>Semester</p>
      <img src={downArrow} alt={"expand"} />
    </FilterButton>
    <FilterButton>
      <p>Level</p>
      <img src={downArrow} alt={"expand"} />
    </FilterButton>
    <FilterButton>
      <p>Credits</p>
      <img src={downArrow} alt={"expand"} />
    </FilterButton>
    <input type="text" placeholder="Name, Keywords, Topics, Etc" />
  </FilterBarContainer>
)

export default FilterBar