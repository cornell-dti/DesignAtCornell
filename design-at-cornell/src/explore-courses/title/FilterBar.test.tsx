import React from 'react';
import renderer from 'react-test-renderer';
import FilterBar from './FilterBar';
import 'jest-styled-components';

describe('<FilterBar />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<FilterBar />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
