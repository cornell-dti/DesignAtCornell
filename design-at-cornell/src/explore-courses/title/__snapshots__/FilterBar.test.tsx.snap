// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterBar /> renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 15%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 p {
  width: 100%;
  font-weight: bold;
  font-size: 16px;
  line-height: 19px;
  text-align: center;
  color: white;
}

.c0 input {
  width: 338px;
  height: 28px;
  border-radius: 43px;
  padding-left: 10px;
  border-style: none;
  background: white;
  background-image: url(magnifying-glass.svg);
  background-position: 7px 7px;
  background-repeat: no-repeat;
  text-indent: 30px;
}

.c0 input:focus {
  outline: none;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  height: 28px;
  background: rgba(255,255,255,0.53);
  border-radius: 43px;
  margin-right: 25px;
  padding-left: 15px;
  padding-right: 15px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 p {
  font-weight: bold;
  font-size: 12px;
  padding-right: 20px;
  color: black;
  margin: 0;
}

.c1 img {
  width: 12px;
  height: 12px;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <p>
      Design Areas
    </p>
    <img
      alt="expand"
      src="down-arrow.png"
    />
  </div>
  <div
    className="c1"
  >
    <p>
      Major/Minors
    </p>
    <img
      alt="expand"
      src="down-arrow.png"
    />
  </div>
  <div
    className="c1"
  >
    <p>
      Semester
    </p>
    <img
      alt="expand"
      src="down-arrow.png"
    />
  </div>
  <div
    className="c1"
  >
    <p>
      Level
    </p>
    <img
      alt="expand"
      src="down-arrow.png"
    />
  </div>
  <div
    className="c1"
  >
    <p>
      Credits
    </p>
    <img
      alt="expand"
      src="down-arrow.png"
    />
  </div>
  <input
    placeholder="Name, Keywords, Topics, Etc"
    type="text"
  />
</div>
`;
